spec/date_range_spec.rb

```diff
@@ -0,0 +1,23 @@
+require_relative 'spec_helper'
+require_relative '../lib/date_range'
+
+describe DateRange do
+  it 'parses date in the correct format %Y-%m-%d' do
+    expect(DateRange.valid_date?('2019-05-01')).must_equal true
+  end
+
+  it 'rejects invalid date' do
+    expect(DateRange.valid_date?('2019-13-01')).must_equal false
+  end
+
+  it 'checks for overlaping date ranges' do
+    expect(DateRange.dates_overlap?(['2019-05-01', '2019-05-02', '2019-05-03'], ['2019-05-01', '2019-05-02', '2019-05-03'])).must_equal true
+    expect(DateRange.dates_overlap?(['2019-05-01', '2019-05-02', '2019-05-03'], ['2019-05-02', '2019-05-03', '2019-05-04'])).must_equal true
+    expect(DateRange.dates_overlap?(['2019-05-01', '2019-05-02', '2019-05-03'], ['2019-05-03', '2019-05-04', '2019-05-05'])).must_equal false
```

Comment:
These are very hard to read! You might break each of these out into a separate test with a descriptive name. This has the added benefit of, if one of them fails, you know exactly which one it was.